src/test/java/com/notably/logic/suggestion/commands/OpenSuggestionCommandTest.java

```diff
@@ -0,0 +1,108 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.commons.core.path.exceptions.InvalidPathException;
+import com.notably.logic.suggestion.model.ModelStub;
+import com.notably.model.Model;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockImpl;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeImpl;
+import com.notably.model.block.Title;
+
+public class OpenSuggestionCommandTest {
+    private static BlockTree blockTree;
+    private static AbsolutePath toRoot;
+    private static AbsolutePath toCs2103;
+    private static AbsolutePath toCs3230;
+    private static AbsolutePath toCs2103Week1;
+    private static AbsolutePath toCs2103Week2;
+    private static AbsolutePath toCs2103Week3;
+    private static AbsolutePath toCs2103Week1Lecture;
+
+    private Model model;
+    private OpenSuggestionCommand openSuggestionCommand;
+
+    @BeforeAll
+    public static void setUp() throws InvalidPathException {
+        toRoot = AbsolutePath.fromString("/");
+        toCs2103 = AbsolutePath.fromString("/CS2103");
+        toCs3230 = AbsolutePath.fromString("/CS3230");
+        toCs2103Week1 = AbsolutePath.fromString("/CS2103/Week1");
+        toCs2103Week2 = AbsolutePath.fromString("/CS2103/Week2");
+        toCs2103Week3 = AbsolutePath.fromString("/CS2103/Week3");
+        toCs2103Week1Lecture = AbsolutePath.fromString("/CS2103/Week1/Lecture");
+    }
+
+    @BeforeEach
+    public void setUpTestTree() {
+        blockTree = new BlockTreeImpl();
+        Block cs2103 = new BlockImpl(new Title("CS2103"));
+        Block cs3230 = new BlockImpl(new Title("CS3230"));
+        Block week1 = new BlockImpl(new Title("Week1"));
+        Block week2 = new BlockImpl(new Title("Week2"));
+        Block week3 = new BlockImpl(new Title("Week3"));
+        Block lecture = new BlockImpl(new Title("Lecture"));
+        blockTree.add(toRoot, cs2103);
+        blockTree.add(toRoot, cs3230);
+        blockTree.add(toCs2103, week1);
+        blockTree.add(toCs2103, week2);
+        blockTree.add(toCs2103, week3);
+        blockTree.add(toCs2103Week1, lecture);
+    }
+
+    @BeforeEach
+    public void initializeModel() {
+        model = new ModelStub();
+    }
+
+    @Test
+    public void constructor_nullPath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new OpenSuggestionCommand(null));
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        OpenSuggestionCommand openSuggestionCommand = new OpenSuggestionCommand(toRoot);
+        assertThrows(NullPointerException.class, () -> openSuggestionCommand.execute(null));
+    }
```

Comment:
You should add at least one more test case for the `execute` method! In fact, if you ask me, I'd say don't need to test the other methods. Simply testing your `execute` method should suffice.

Consider this:
- You can test your `getPossiblePaths` method or your `getSuggestions` method all day and make sure it works. Nevertheless, if you never test your `execute` method thoroughly, you can never confidently say that your code works, since the user of your module uses the `execute` method only!
- On the other hand, if you test the `execute` method only, you can always be confident that your `execute` method, plus all the methods it uses (well, perhaps not all) works!